apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -291,20 +327,13 @@ def write(self, values: pa.Table) -> None:
                 attr_cols_map[name] = values.column(name).to_pandas()
         assert n is not None
 
-        dim_cols_list = [list(dim_col) for dim_col in dim_cols_list]
+        dim_cols_tuple = tuple([list(dim_col) for dim_col in dim_cols_list])
```

Comment:
this wastes memory / cpu by converting to list then to a tuple.  Better form:
```
dim_cols_tuple = tuple(list(dim_col) for dim_col in dim_cols_list)
```

Also, per the docs, TileDB-Py.__setitem__ already accepts a numpy.ndarray, so you don't need to convert it to a list, so ultimately:
```
dim_cols_tuple = tuple(dim_cols_list)
```

We need to be careful with all this extra alloc & copy activity, as it can get expensive and generates GC pressure.